### ISSUE TITLE
[feat] removed `jsx-uses-vars` would still be useful for MDX; requesting undeletion

### ISSUE THREAD

**Issue by controversial** [Rel1cx/eslint-react#1618]
### Problem Description

Even though eslint 10 added native support for tracking JSX references, it seems like `eslint-plugin-mdx` is not (yet) working with this native support; mdx files still have false-positives for no-unused-vars when they’re used by JSX expressions. 

### Alternative Solutions

Maybe this could be fixed upstream in eslint-plugin-mdx; I [left a comment](https://github.com/mdx-js/eslint-mdx/issues/444#issuecomment-4057714745) there.

### Feature Details

However, it would be useful to still have `jsx-uses-vars` available to use in configs for cases like this (even if no longer enabled by default). I would still find the rule useful for MDX even though plain JSX doesn’t require it in ESLint 10.

### Examples

```mdx
import { MyComponent } from "component/path";

This _markdown_ file renders <MyComponent />
```

### Evaluation Checklist

- [x] I have had problems that this feature would solve
- [x] I could not find a way to solve the problem with existing features or workarounds
- [x] I have thought very hard about potential edge cases and downsides, and they are acceptable
- [x] I think the feature is well-defined and would provide clear value to users

**Comment by controversial** [comment]
Just kidding, everything works properly if I add
```ts
    {
      files: ["**/*.mdx"],
      languageOptions: {
        parserOptions: {
          ecmaFeatures: { jsx: true },
        },
      },
    },
```
